Tools/Scripts/webkitpy/layout_tests/controllers/layout_test_runner.py

```diff
@@ -512,6 +531,9 @@ def __init__(self, name, test_inputs):
         self.test_inputs = test_inputs
         self.needs_servers = test_inputs[0].needs_servers
 
+    def __contains__(self, s):
```

Comment:
I still feel like this is a surprising implementation of `__contains__`.

In my mind, it makes sense for `TestShard` to be a [`Collection`](https://docs.python.org/3/library/collections.abc.html#collections.abc.Collection), with an iterator like `def __iter__(self): return iter(self.test_inputs)`, and note the default implementation of [`__contains__`](https://docs.python.org/3/reference/datamodel.html#object.__contains__) which just checks if `item` is in the iterator.

As such, I'd expect something like `def __contains__(self, s: TestInput)`, rather than `def __contains__(self, s: str)`.